logstash-core/src/main/java/org/logstash/instrument/witness/PipelinesWitness.java

```diff
@@ -27,4 +36,46 @@ public PipelineWitness pipeline(String name) {
         return pipelines.computeIfAbsent(name, k -> new PipelineWitness(k));
     }
 
+    @Override
+    public void genJson(JsonGenerator gen, SerializerProvider provider) throws IOException {
+        new Serializer().innerSerialize(this, gen, provider);
+    }
+
+    /**
+     * The Jackson serializer.
+     */
+    public static class Serializer extends StdSerializer<PipelinesWitness> {
+
+        /**
+         * Default constructor - required for Jackson
+         */
+        public Serializer() {
+            this(PipelinesWitness.class);
+        }
+
+        /**
+         * Constructor
+         *
+         * @param t the type to serialize
+         */
+        protected Serializer(Class<PipelinesWitness> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(PipelinesWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeStartObject();
+            innerSerialize(witness, gen, provider);
+            gen.writeEndObject();
+        }
+
+        void innerSerialize(PipelinesWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeObjectFieldStart(KEY);
+            for (Map.Entry<String, PipelineWitness> entry : witness.pipelines.entrySet()) {
```

Comment:
I gave it a quick try, and ran into this error:
`"No ObjectCodec defined for the generator, can only serialize simple wrapper types (type passed java.util.concurrent.ConcurrentHashMap)"`

A bit of googling, and it was suggested to add ` gen.setCodec(new ObjectMapper());` ...however, the output is bit off (fixable), but seems odd to have to create an additional ObjectMapper since for the (asJson) workflows I am already explicitly creating one. 

It's only a couple extra lines of code, and would prefer to avoid mixing the explicit asJson serialization with the annonation driven approach. I have tests in this PR that confirm that they equal.